lymph/unilateral.py

```diff
@@ -948,9 +961,8 @@ def _evolve(
 
         return state_probs
 
-    #right now the micro_mod and the growth parameter can be set differently for each edge
-    #If we want only 1 micro_mod and growth_parameter to be available, we need to set them together
-    #Or add an option to treat them separately or not (important for sampling)
+    # Not sure whether we want to keep the option here to assign parameters as an array
+    # if so, I would implement the function to handle dictionary inputs and array inputs.

```

Comment:
Ideally, we have both: Let the user either pass the new spread probs as keyword arguments to the `assign_parameters` method OR as an array/list of values. A possible implementation could look like this:

```python
def assign_parameters(self, new_params_list = None, **new_params_kwargs):
    """Set the system's new spread probs..."""
    if new_params_list is not None:
        # This might actually be stupic, because dicts don't guarantee the order of their entries...
        for setter_func, new_param in zip(self._setter_lookup.values(), new_params_list):
            setter_func(new_param)

    # Then comes what you already implemented
```
That way one can call the function with a list of values like so:
```python
system.assign_parameters([0.2, 0.3, 0.6, ...])
```
And even override selected values from the passed array by providing a new value in a keyword argument.

Reply:
I would not assign the new_params_list through the self._setter_lookup directly by looping, since as you said, the 
dictionary only guarantees the order since python 3.7. If we can guarantee that the user is using python 3.7+ we can just loop through the lookup dictionary. 
Nonetheless, the order of the dictionary was not fixed based on the order we had before to assign parameters. (since I assumed that we are not considering order anyways). Hence the input list would need to be ordered based on the `_setter_lookup`. If we can guarantee that, your approach would be fine. I would have used the still existing `spread_prob`setter.  Which is of course less elegant since we need more code to do the same thing.